src/main/java/billgates/usecases/insert_entry/InsertEntryUseCase.java

```diff
@@ -0,0 +1,18 @@
+package billgates.usecases.insert_entry;
+
+import billgates.database.QueryEntryData;
+import billgates.interface_adapters.DatabaseGateway;
+
+public class InsertEntryUseCase implements InsertEntryInputPort{
+
+    private final DatabaseGateway gateway;
+
+    public InsertEntryUseCase(DatabaseGateway gateway){
+        this.gateway = gateway;
+    }
+
+    @Override
+    public void insertEntry(int billId, QueryEntryData entry) {
+        gateway.insertEntry(billId, entry);
```

Comment:
This use case is more complex than you expected.

There should be an `InsertEntryRequestModel` containing the required information of the newly created entry. 

Then, in the method, you should construct a `QueryEntryData` and pass it in the `Gateway#insertEntry`.